server.js

```diff
@@ -36,7 +41,22 @@ app.get('*', (req, res) => {
     }
   });
 });
-app.listen(3003, 'localhost', function(err) {
+
+app.post('/submit-email-form', (req, res) => {
+  const { email } = req.body;
+  const result = validateEmailForm(email);
+
+  if (result.valid === true) {
+    // here you would save this to a database, or whatever you want
+    // db.addEmail(email)
+
+    res.redirect(`/about?success=true&value=${email}`);
+  } else {
+    res.redirect(`/about?errors=${result.errors.join(',')}&success=false&value=${email}`);
+  }
+});
```

Comment:
this is the route that the form will POST to if there's no JS. Notice how it's tiny - the logic is encapsulated in `validateEmailForm`.